### ISSUE TITLE
[REVIEW]: deeplenstronomy: A dataset simulation package for strong gravitational lensing

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2854]
**Submitting author:** @rmorgan10 (<a href="http://orcid.org/0000-0002-7016-5471">Robert Morgan</a>)
**Repository:** <a href="https://github.com/deepskies/deeplenstronomy" target ="_blank">https://github.com/deepskies/deeplenstronomy</a>
**Version:** v0.0.1.2
**Editor:** @pdebuyl
**Reviewer:** @shreyasbapat, @jiwoncpark
**Archive:** <a href="https://doi.org/10.5281/zenodo.4479712" target="_blank">10.5281/zenodo.4479712</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/e978dd566d1f290055a02d76288e95e1/status.svg)](https://joss.theoj.org/papers/e978dd566d1f290055a02d76288e95e1)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/e978dd566d1f290055a02d76288e95e1"><img src="https://joss.theoj.org/papers/e978dd566d1f290055a02d76288e95e1/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/e978dd566d1f290055a02d76288e95e1/status.svg)](https://joss.theoj.org/papers/e978dd566d1f290055a02d76288e95e1)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/deepskies/deeplenstronomy" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@shreyasbapat & @jiwoncpark, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @pdebuyl know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/deepskies/deeplenstronomy">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@rmorgan10) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/deepskies/deeplenstronomy">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@rmorgan10) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @shreyasbapat, @jiwoncpark it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by pdebuyl** [comment]
@shreyasbapat @jiwoncpark make sure to accept the invitation to the reviewers group and to have a look at the reviewer guidelines linked to at the top of this review page.

The review process will happen in this issue page, so questions to the author or to me can be added as comments here.

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02854/joss.02854/10.21105.joss.02854.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02854/joss.02854/10.21105.joss.02854.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1038/s41586-020-2649-2 is OK
- 10.1109/MCSE.2007.55 is OK
- 10.1051/0004-6361/201322068 is OK
- 10.1086/301513 is OK
- 10.1103/PhysRevD.98.043526 is OK
- 10.3847/1538-4357/ab042c is OK

MISSING DOIs

- 10.25080/majora-92bf1922-00a may be a valid DOI for title: Data structures for statistical computing in python
- 10.1016/j.dark.2018.11.002 may be a valid DOI for title: Lenstronomy: multi-purpose gravitational lens modelling software package

INVALID DOIs

- None
```

**Comment by shreyasbapat** [comment]
There are a few DOIs missing !

**Comment by rmorgan10** [comment]
> There are a few DOIs missing !

The `pandas` DOI has been added, and the `lenstronomy` citation has been updated to the journal article (including the DOI) instead of ArXiv.

**Comment by pdebuyl** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1038/s41586-020-2649-2 is OK
- 10.25080/majora-92bf1922-00a is OK
- 10.1109/MCSE.2007.55 is OK
- 10.1016/j.dark.2018.11.002 is OK
- 10.1051/0004-6361/201322068 is OK
- 10.1086/301513 is OK
- 10.1103/PhysRevD.98.043526 is OK
- 10.3847/1538-4357/ab042c is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @jiwoncpark, please update us on how your review is going.

**Comment by whedon** [comment]
:wave: @shreyasbapat, please update us on how your review is going.

**Comment by pdebuyl** [comment]
@jiwoncpark @shreyasbapat gentle reminder

**Comment by jiwoncpark** [comment]
@pdebuyl Apologies for the delay in my update. I should be finished with my review by tomorrow.

**Comment by pdebuyl** [comment]
Thank you for the reply @jiwoncpark . Since the spring, the timeframe for a JOSS review is 6 weeks maximum. I update the review regularly to make sure that the review will happen but there is no emergency for tomorrow.

**Comment by pdebuyl** [comment]
@shreyasbapat @jiwoncpark gentle reminder

There are several ways to proceed with the review. You can check the items and proceed with direct feedback to the author or prepare a bulk review. Either way is fine, feel free to ask questions here about it.

**Comment by rmorgan10** [comment]
@pdebuyl I would like to put the `deeplenstronomy` paper on arXiv at some point, but this is the first time I've worked with JOSS. Would you recommend waiting until after the review process for any reason?

**Comment by pdebuyl** [comment]
Hi @rmorgan10 I would recommend waiting for the publication here to save you the trouble of updating the arXiv version. You are free to proceed however (see https://joss.readthedocs.io/en/latest/submitting.html?highlight=arxiv#preprint-policy ).

Just make sure to not reference the paper as published in joss yet (mention of submission is fine of course).

**Comment by rmorgan10** [comment]
@pdebuyl If I wait until after the review process (and `deeplenstronomy` is accepted), does JOSS release the TeX for the paper to me?

**Comment by pdebuyl** [comment]
No. Officially, JOSS only sees the input .md file and the pdf output. It is possible to execute our code locally of course. It uses pandoc. It is also possible to use pandoc directly, provided you have the template and logo. I can help if you wish (not anymore today however).

**Comment by arfon** [comment]
@rmorgan10 - you might be able to make a TeX version using this script: https://github.com/mattpitkin/psrqpy/blob/master/paper/Makefile

**Comment by pdebuyl** [comment]
@shreyasbapat @jiwoncpark any news on this?

**Comment by shreyasbapat** [comment]
I promise to review this in next 2 days😀

**Comment by jiwoncpark** [comment]
Hi @rmorgan10, I created an [issue](https://github.com/deepskies/deeplenstronomy/issues/41#issue-768442817) at the `deeplenstronomy` repo to address some of the review checklist items, including tests and function docstrings. Separately, I'm adding my comments on the paper manuscript below. @pdebuyl Please let me know if the issue or paper comments should be placed elsewhere.

**Paper comments**

*Background*
- "They also provided critical evidence during tests of the theory of General Relativity." --> would be better in present tense (SL as a standing test of GR and modified theories of gravity)
- "SDSS, DES, and LSST" --> I'm not sure about the JOSS rules on survey acronyms, but I wonder if these should be written out.
- You can introduce the 2D-CNN acronym after "two-dimensional convolutional neural network" but see my comments below.
- "an integral parts" --> "an integral part"

*Statement of need*
- "SL detection" : My impression is that this software is more general and can be used for lens modeling, time series analysis, etc. not just detection.
- I wouldn't restrict the architecture to 2D CNNs either, to make room for, e.g. graph-based architectures.

*Summary*
- "data set" --> "dataset" (multiple places)
- The second paragraph contains a long list, some elements of which don't quite correspond to "[making the] datasets as realistic as possible." Maybe break it up into several logically coherent lists? You could present "inspect and visualize the simulation outputs," "simulate time series data" as additional tools and functionalities, respectively. 
- Can you elaborate on the time series simulation portion? You mainly discuss images before this point. After looking at a demo notebook, I understand this to mean a time series of images?
- A code snippet and a short yaml file would be helpful, as a preview of the API.

*State of the field*
Currently, there is no section or paragraph describing the "state of the field" (addressing the review criterion: "Do the authors describe how this software compares to other commonly-used packages?"). I'm not sure if this part of the JOSS requirements, but if I can mention my own software package, `baobab` (https://github.com/jiwoncpark/baobab) is a very similar `lenstronomy` wrapper with a focus on lens modeling. It was used to generate the training set for two modeling papers, [Park et al 2020](https://arxiv.org/abs/2012.00042) and [Wagner-Carena et al 2020](https://arxiv.org/abs/2010.13787). `deeplenstronomy` seems more general.

*Overall style comments*
- Sentences tend to be quite long, with one or more descriptive clauses. I recommend splitting them into shorter sentences.
- Using active voice and fewer nouns would also help with the above.
Example: "Due to the inherent dependence of the performance of ML approaches on their training data,the 2D-CNN approach to SL detection is in tension with scientific reproducibility without a clear prescription for the simulation of the training data." --> Because supervised ML approaches depend heavily on the training data, the principle of reproducibility dictates that they be accompanied by a clear ... (or something like that)

**Comment by pdebuyl** [comment]
Thank you @jiwoncpark for the review! The issue and comments are at the right place. There is some freedom with respect to how the review is conducted, as long as I see the progress on this issue page.

**Comment by pdebuyl** [comment]
@rmorgan10 there is some confusion related to the link error mentioned in https://github.com/deepskies/deeplenstronomy/issues/41 : you have a repository called deeplenstronomy with the notebooks. Is that repository supposed to remain online? If so, could you rename it?

**Comment by pdebuyl** [comment]
Edit/PS to previous comment: "you have a repository called deeplenstronomy in your github profile"

**Comment by rmorgan10** [comment]
> Edit/PS to previous comment: "you have a repository called deeplenstronomy in your github profile"

Yeah that was a staging area for the notebooks before we enabled github-pages for the deepskies/deeplenstronomy repo. I should really just delete the personal repo. Thanks for pointing that out!

**Comment by rmorgan10** [comment]
@jiwoncpark Thanks for your review! I think you've pointed out things that will help improve `deeplenstronomy` a lot. I've started working on some of the small things in the issue you opened in the `deeplenstronomy` repo.

I just wanted to give everyone a heads up that I won't be focusing too much on implementing major changes from the JOSS reviews until I have some more free time (which for me will start Jan 5). So @shreyasbapat no need to rush to get your review in if you are also busy.

Thanks again!

**Comment by pdebuyl** [comment]
Hi @rmorgan10 thanks for the heads up. I'll check back in two weeks as well.

**Comment by pdebuyl** [comment]
@rmorgan10 gentle reminder

**Comment by rmorgan10** [comment]
Hi @pdebuyl, thanks for the ping. I'll be wrapping up my responses @jiwoncpark 's review this week. I believe I'm still waiting on @shreyasbapat 's review though. If possible, I'd like to work on them simultaneously, so @shreyasbapat do you have an estimate for when your comments will be ready?

**Comment by pdebuyl** [comment]
Thanks for the reply @rmorgan10 . Indeed, you can wait for the second review (JOSS allows for both ways of working).

**Comment by pdebuyl** [comment]
@shreyasbapat gentle reminder for the review.

The review page was opened more than 6 weeks ago, so it would be nice to have the review soon.

**Comment by shreyasbapat** [comment]
I think @jiwoncpark mentioned a majority of issues already. I created an issue which might not be very imporant from JOSS persective, but from a user's and maintainability perspective. 

I feel once the tests and documentation is done, we would be good to go.

https://github.com/deepskies/deeplenstronomy/issues/42

**Comment by pdebuyl** [comment]
Hi @shreyasbapat thank you for the feedback.

**Comment by pdebuyl** [comment]
@rmorgan10 many checkboxes have been ticked by @shreyasbapat now.

I suggest that you take into account all the "non paper" feedback so far. @shreyasbapat , your comments on the paper will be needed as well.

**Comment by rmorgan10** [comment]
Hi All, thanks for the reviews! If I understand correctly, it looks like the major required improvements are:

1. Making the doc strings available via sphinx
2. Organize the tests and clarify what is being tested
3. Minor edits to the paper text

I am happy to do all three and will hopefully have all issues resolved at the end of next week if not sooner. I am glad the code and its functionalities were satisfactory. Thanks again for your careful reviews of deeplenstronomy!

**Comment by pdebuyl** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02854/joss.02854/10.21105.joss.02854.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02854/joss.02854/10.21105.joss.02854.pdf) :page_facing_up: :point_left:

**Comment by pdebuyl** [comment]
>    - A code snippet and a short yaml file would be helpful, as a preview of the API.

Including such content in the paper goes against the guidelines, we consider it as documentation and recommend its inclusion in the proper documentation of the package.

**Comment by rmorgan10** [comment]
Thanks @pdebuyl . Code snippets and yaml file excerpts are available in the provided example notebooks, which are referenced in the paper, so hopefully this comment is already addressed within the JOSS guidelines.

https://deepskies.github.io/deeplenstronomy/Notebooks/

**Comment by rmorgan10** [comment]
Hi All,

I just wanted to give an update that I've finished responding to the initial reviews. Please let me know when you have feedback for my responses

**Comment by pdebuyl** [comment]
Thank you for the notification @rmorgan10

@shreyasbapat @jiwoncpark can you go over the changes and update the opened issues at deeplenstronomy's repo and the checkboxes here?

**Comment by shreyasbapat** [comment]
Looks good from my side

**Comment by pdebuyl** [comment]
Hi @shreyasbapat thanks for finishing the review!

Can the issue https://github.com/deepskies/deeplenstronomy/issues/42 be closed?

**Comment by jiwoncpark** [comment]
The text and code look really good to me, @rmorgan10! I've checked off the boxes. Super minor suggestions/comments:

- In line 43, I believe the package name ```SLSprinkler``` doesn't have a space and how about "imputes" --> "adds?" I'm also confirming the correct citation for the package, but this shouldn't hold you back in the JOSS review.
- Just FYI, you can choose what files to ignore in test coverage. Useful for excluding config files, files consisting of external modules tested elsewhere
- You might want to consider setting up GH actions for CI.

Once again, congrats on  a very nice package!

**Comment by rmorgan10** [comment]
Thanks @jiwoncpark! I've implemented your line 43 changes. Please let me know if there is a better citation for `SLSprinkler` or `baobab`. I am considering setting up GitHub actions for CI at some point, but that will be slightly down the road.

**Comment by pdebuyl** [comment]
Thank you @jiwoncpark for the review!

@rmorgan10 I'll now proceed to some editorial checks before handing out the paper to the editor-in-chief (see https://joss.readthedocs.io/en/latest/editing.html#after-reviewers-recommend-acceptance ).

**Comment by pdebuyl** [comment]
@whedon check references

**Comment by pdebuyl** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1038/s41586-020-2649-2 is OK
- 10.25080/majora-92bf1922-00a is OK
- 10.1109/MCSE.2007.55 is OK
- 10.1016/j.dark.2018.11.002 is OK
- 10.1051/0004-6361/201322068 is OK
- 10.1086/301513 is OK
- 10.1103/PhysRevD.98.043526 is OK
- 10.3847/1538-4357/ab042c is OK

MISSING DOIs

- 10.1163/1872-9037_afco_asc_3175 may be a valid DOI for title: baobab

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02854/joss.02854/10.21105.joss.02854.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02854/joss.02854/10.21105.joss.02854.pdf) :page_facing_up: :point_left:

**Comment by rmorgan10** [comment]
> MISSING DOIs
>
>- 10.1163/1872-9037_afco_asc_3175 may be a valid DOI for title: baobab

I don't think this doi is correct, so I'm not too worried about it appearing under the "Missing DOIs" section

**Comment by pdebuyl** [comment]
Hi @rmorgan10 don't worry about the DOI. I do a manual check of the DOIs, the automatic feature of whedon is a helpful starting point but not the source of truth :-)

**Comment by jiwoncpark** [comment]
Hi, please use this citation for `baobab`:
```
Ji Won Park. (2021, January 28). jiwoncpark/baobab: v0.1.2 (Version v0.1.2). Zenodo. http://doi.org/10.5281/zenodo.4476822
```

**Comment by rmorgan10** [comment]
Updated! Thanks!

**Comment by rmorgan10** [comment]
@pdebuyl Here is the version number and archive DOI for the updated release of deeplenstronomy:

Version: v0.0.1.2
DOI: 10.5281/zenodo.4479712

**Comment by jiwoncpark** [comment]
If it's not too late, here is the Zenodo for `SLSprinkler`: http://doi.org/10.5281/zenodo.4480392

**Comment by rmorgan10** [comment]
@jiwoncpark I've updated the bibliography entry for SLSprinkler, thanks for the DOI :)

**Comment by pdebuyl** [comment]
Hi @rmorgan10 

In the paper, can you use explicity journal names instead of the bibtex macro. `\prd` -> `Physical Review D`, etc? This is for Abbott,  Astropy, Ivezić, York.

SciPy has a paper out, please update the citation. see https://www.scipy.org/citing.html

**Comment by pdebuyl** [comment]
@whedon set 10.5281/zenodo.4479712 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4479712" target="_blank">10.5281/zenodo.4479712</a> is the archive.

**Comment by pdebuyl** [comment]
@whedon set v0.0.1.2 as version

**Comment by whedon** [comment]
OK. v0.0.1.2 is the version.

**Comment by rmorgan10** [comment]
@pdebuyl I've updated the bibliography with your recommendations, thanks! https://github.com/deepskies/deeplenstronomy/commit/032f6a5aa7aee1df7d4b6e6d1851304dde8c05ab

**Comment by pdebuyl** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02854/joss.02854/10.21105.joss.02854.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02854/joss.02854/10.21105.joss.02854.pdf) :page_facing_up: :point_left:

**Comment by pdebuyl** [comment]
Thank you. Can you edit the archive's title to match the paper? It should read "deeplenstronomy: A dataset simulation package for strong gravitational lensing"

**Comment by rmorgan10** [comment]
Yup! Sorry about that, it's now updated

**Comment by pdebuyl** [comment]
Thanks :-) The paper will now go the the editor in chief.

Thank you @shreyasbapat and @jiwoncpark , with special mention for the text changes proposed by @jiwoncpark :-)

**Comment by pdebuyl** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1038/s41586-020-2649-2 is OK
- 10.5281/zenodo.4480392 is OK
- 10.5281/zenodo.4476822 is OK
- 10.1038/s41592-019-0686-2 is OK
- 10.25080/majora-92bf1922-00a is OK
- 10.1109/MCSE.2007.55 is OK
- 10.1016/j.dark.2018.11.002 is OK
- 10.1051/0004-6361/201322068 is OK
- 10.1086/301513 is OK
- 10.1103/PhysRevD.98.043526 is OK
- 10.3847/1538-4357/ab042c is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2071

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2071, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by pdebuyl** [comment]
@rmorgan10 please check the final pdf https://raw.githubusercontent.com/openjournals/joss-papers/7b45a09803a630144e588ffe8f36aa033e3dbd66/joss.02854/10.21105.joss.02854.pdf 

The editor-in-chief in rotation will pick up the process from here. Thanks for submitting to JOSS :-)

**Comment by rmorgan10** [comment]
It looks good to me! Thanks @pdebuyl, @jiwoncpark, and @shreyasbapat for your reviews and for helping `deeplenstronomy` get to this point!

**Comment by kyleniemeyer** [comment]
Looks good to me, publishing now!

**Comment by kyleniemeyer** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1357137865742909442) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2073
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02854](https://doi.org/10.21105/joss.02854)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kyleniemeyer** [comment]
Congrats @rmorgan10 on your article's publication in JOSS!

Many thanks to @shreyasbapat and @jiwoncpark for reviewing this, and @pdebuyl for editing.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02854/status.svg)](https://doi.org/10.21105/joss.02854)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02854">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02854/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02854/status.svg
   :target: https://doi.org/10.21105/joss.02854
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02854/status.svg)](https://doi.org/10.21105/joss.02854)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
